kartothek/io/dask/_update.py

```diff
@@ -30,32 +51,37 @@ def _update_dask_partitions_shuffle(
     if ddf.npartitions == 0:
         return ddf
 
-    splits = np.array_split(
-        np.arange(ddf.npartitions), min(ddf.npartitions, num_buckets)
+    if num_buckets is not None:
+        if isinstance(sort_partitions_by, list):
+            bucket_by = sort_partitions_by
+        elif isinstance(sort_partitions_by, str):
+            bucket_by = [sort_partitions_by]
+        else:
+            bucket_by = None
```

Comment:
this should raise an `ValueError`. The type is neither a list nor a string, but they user might have used a somewhat valid type (e.g. tuple) or something weird (like a random object) and we're silently ignoring it.

Reply:
Promoted the bucket_by to a user input as discussed above https://github.com/JDASoftwareGroup/kartothek/pull/168#discussion_r347443451